src/extension.ts

```diff
@@ -79,23 +79,68 @@ let decorationType: TextEditorDecorationType = window.createTextEditorDecoration
   }
 );
 
+const jdk8 = "1.8.232";
+const jdk8Action = "JDK8";
+
+const jdk11 = "1.11";
+const jdk11Action = "JDK11";
+
+const default_jabba_ver = "0.11.2";
+
+const config = workspace.getConfiguration("metals");
+
+const java9 = "Java9";
 export async function activate(context: ExtensionContext) {
   detectLaunchConfigurationChanges();
   checkServerVersion();
 
   getJavaHome()
     .then(javaHome => fetchAndLaunchMetals(context, javaHome))
     .catch(err => {
-      const message =
-        "Unable to find a Java 8 or Java 11 installation on this computer. " +
-        "To fix this problem, update the 'Java Home' setting to point to a Java 8 or Java 11 home directory";
+      var message = "";
+      if (err.message === java9) {
```

Comment:
nit: can we extract this if/else chain into a method so we can do `const message = getMessage(err) + "To fix this problem ..."`

Reply:
I agree, this would also allow avoiding mutation.

By the way, @sswistun-vl even if we need a mutable variable, I recommend using `let` instead of `var`, since `let` has way more intuitive and sensible scoping rules (`let` is block-scoped whereas `var` is function-scoped)

Reply:
Changed, good to know about var and let difference, I don't know too much about js/ts.